sc-locked-asset-factory/src/lib.rs

```diff
@@ -43,15 +50,17 @@ pub trait LockedAssetFactory:
     fn whitelist(&self, address: Address) -> SCResult<()> {
         only_owner!(self, "Permission denied");
 
-        self.whitelisted_contracts().insert(address);
+        let is_new = self.whitelisted_contracts().insert(address);
+        require!(is_new, "Address already whitelisted");
```

Comment:
There's no reason to return error if the address was already in the whitelist, as it does not harm :)

Reply:
I know, but it seems it helps RV with their modeling